opendevin/controller/agent_controller.py

```diff
@@ -126,7 +131,9 @@ async def report_error(self, message: str, exception: Exception | None = None):
     async def add_history(self, action: Action, observation: Observation):
         if isinstance(action, NullAction) and isinstance(observation, NullObservation):
             return
-        self.state.history.append((action, observation))
+        logger.debug(
+            f'Adding history ({type(action).__name__} with id={action.id}, {type(observation).__name__} with id={observation.id})'
+        )
```

Comment:
Should remove this `add_history` method, either in this PR or follow-up. It was still here just for testing, to help ensure that the behavior is the same. (events displayed here, but no longer appended here, match those accessible via history)